docs/eslint-plugin-mlflow-docs/rules/valid-notebook-url.js

```diff
@@ -0,0 +1,178 @@
+const fs = require("fs");
+const path = require("path");
+
+/**
+ * ESLint rule to validate NotebookDownloadButton URLs.
+ *
+ * This rule ensures that NotebookDownloadButton components have valid MLflow repository URLs
+ * that point to existing files in the repository.
+ *
+ * @type {import('eslint').Rule.RuleModule}
+ */
+module.exports = {
+  meta: {
+    type: "problem",
+    docs: {
+      description: "Detect NotebookDownloadButton with invalid MLflow repository URLs",
+      category: "Possible Errors",
+    },
+    fixable: null,
+    schema: [],
+    messages: {
+      missingHref: "NotebookDownloadButton is missing href attribute",
+      emptyHref: "NotebookDownloadButton href is empty",
+      invalidFormat:
+        'NotebookDownloadButton href must start with "https://raw.githubusercontent.com/mlflow/mlflow/master/"',
+      fileNotFound: 'NotebookDownloadButton href points to non-existent file: "{{path}}"',
+    },
+  },
+
+  /**
+   * Creates the rule implementation.
+   *
+   * @param {import('eslint').Rule.RuleContext} context - The ESLint rule context
+   * @returns {import('eslint').Rule.RuleListener} The rule visitor methods
+   */
+  create(context) {
+    return {
+      /**
+       * Validates JSX opening elements for NotebookDownloadButton components.
+       *
+       * @param {import('estree').Node} node - The JSX opening element node
+       */
+      JSXOpeningElement(node) {
+        // Check if it's <NotebookDownloadButton >
+        if (
+          node.type !== "JSXOpeningElement" ||
+          node.name.type !== "JSXIdentifier" ||
+          node.name.name !== "NotebookDownloadButton"
+        ) {
+          return;
+        }
+
+        // Find href attribute
+        const hrefAttr = node.attributes.find(
+          (attr) => attr.type === "JSXAttribute" && attr.name.name === "href"
+        );
+
+        if (!hrefAttr) {
+          context.report({
+            node,
+            messageId: "missingHref",
+          });
+          return;
+        }
+
+        // Get href value
+        const hrefValue = getHrefValue(hrefAttr);
+
+        if (!hrefValue) {
```

Comment:
The docs wouldn't build if this was missing

Reply:
It won't but that doesn't mean this is useless.